thoth/storages/graph/dgraph.py

```diff
@@ -2029,6 +2053,79 @@ def sync_analysis_result(self, document: dict) -> None:
+                    python_file_digests = PythonFileDigest.from_properties(
+                        sha256=content["sha256"],
+                    ).get_or_create(self.client)
+
+                    FoundFile.from_properties(
+                        source=python_artifact,
+                        target=python_file_digests,
+                        file_path=filepath + content["name"],
+                    ).get_or_create(self.client)
+
+    @enable_vertex_cache
+    def sync_package_analysis_result(self, document: dict) -> None:
+        """Sync the given package analysis result to the graph database."""
+        package_analysis_document_id = PackageAnalysisResultsStore.get_document_id(document)
+        package_name = document["metadata"]["arguments"]["python"]["package_name"]
+        package_version = document["metadata"]["arguments"]["python"]["package_version"]
+        index_url = document["metadata"]["arguments"]["python"]["index_url"]
+
+        # TODO: capture errors on package analysis? result of package-analyzer should be a JSON with error flag
+        package_analyzer_run = PackageAnalyzerRun.from_properties(
+            package_analysis_document_id=package_analysis_document_id,
+            package_analysis_datetime=document["metadata"]["datetime"],
+            package_analyzer_version=document["metadata"]["analyzer_version"],
+            package_analyzer_name=document["metadata"]["analyzer"],
+            debug=document["metadata"]["arguments"]["thoth-package-analyzer"]["verbose"],
+            package_analyzer_error=False,
+            duration=None,  # TODO: assign duration
+        ).get_or_create(self.client)
+
+        package_index = PythonPackageIndex.query_one(self.client, url=index_url)
+        if not package_index:
+            raise PythonIndexNotRegistered(
+                f"Cannot insert PythonPackageVersionEntity record into database, "
+                f"no Python index with url {index_url} registered"
+            )
+
+        entity, _ = self.create_python_package_version_entity(package_name, package_version, index_url)
+        ProvidedBy.from_properties(
+            source=entity,
+            target=python_index,
+        ).get_or_create(self.client)
+
+        InstalledFrom.from_properties(
+            source=entity,
+            target=package_analyzer_run,
+        ).get_or_create(self.client)
+
+        for artifact in document["result"][index_url]["files"]:
+            python_artifact = PythonArtifact.from_properties(
+                artifact_hash_sha256=artifact["sha256"],
+                artifact_name=artifact["name"],
+            ).get_or_create(self.client)
+
+            HasArtifact.from_properties(
+                source=package_analyzer_run,
+                target=python_artifact,
+            ).get_or_create(self.client)
+
+            self._python_file_digests_sync_package_analysis_result(artifact["contents"], python_artifact)
+
```

Comment:
As our result is in json format, to create **filepath** we need to recursively loop and create them. @fridex Here, as you see we are creating filepaths, however why we didn't create them in *package-analyzer* job itself?

Reply:
Nice, +1 on my side. That solution will be much nicer!

Reply:
@fridex If you agree, then I will start changing the output of result to something like:
```json
{
"metadata":{},
"result": {
"https://pypi.org/simple": {
      "files": [
{
"contents": [
{
"filepath": "some_dir/some_file1.py",
"sha256": "some_hash1"
},
{
"filepath": "some_dir/some_file2.py",
"sha256": "some_hash2"
}
],
 "name": "tensorflow-1.14.0-cp27-cp27m-macosx_10_11_x86_64.whl",
"sha256": "46fc216db780b1a7b7a28cf6536d6ec0a171ef2fa546a24934ffe077c734c891"
},
{}
]
}
}
}
```

Reply:
Yes, that looks like the right solution :+1: :1st_place_medal: